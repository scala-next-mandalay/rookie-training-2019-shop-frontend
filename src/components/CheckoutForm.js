import React from 'react';
import {Button,Grid,Divider,Box,Paper,TextField} from '@material-ui/core';
import TitleBar from '../containers/TitleBar';
import { FormattedMessage } from 'react-intl';
import CartTable from '../containers/CartTable';
import { makeStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import { validateForm } from '../util';
import clsx from 'clsx';
import '../assets/style.css';
import { withRouter } from 'react-router-dom';
import HomeIcon from '@material-ui/icons/Home';

const useStyles = makeStyles(theme => ({
  paper: {
    padding: theme.spacing(2),
    marginTop: theme.spacing(2),
  },
  submitButton: {
    margin: theme.spacing(3, 0)
  },
  boxTitle: {
    fontWeight: 600,
    padding: theme.spacing(2, 0),
    marginLeft: theme.spacing(3)
  },
  boxValue: {
    padding: theme.spacing(2, 0),
    letterSpacing: '2px',
     marginLeft: theme.spacing(2)
  },
  main :{
    marginTop: theme.spacing(5),
  },
    gridPaper: {
   backgroundColor: '#f2f2f2',
  },
  gridTotalPrice :{
    backgroundColor: '#fff',
    borderRadius:'2px',
    border : '1px solid #f2f2f2',
    marginRight: '40px',
    textAlign:'center'
  },
  textField: {
    backgroundColor : '#fff',
    border: '1px solid #f2f2f2'
  },
  dense: {
    marginTop: theme.spacing(2),
  },
   title: {
    marginLeft: theme.spacing(2),
    flex: 1,
  },
  toolbar:{
  marginTop: theme.spacing(10),
  },
  homeSpace: {
    marginRight: theme.spacing(1),
  }
 
}));

const CheckoutForm = ({ cart,setRequestParams, history,totalPrice,totalQuantity}) => {
  const classes = useStyles();
  
  const validationSetting = {
    isEmpty: ['first_name', 'address1', 'country', 'city']
  };
    
  const state  = {
    first_name: "", 
    last_name: "", 
    address1: "",
    address2: "",
    country: "",
    state: "",
    city: "",
  };
  
  const [form, setForm] = React.useState(state );
  
  const [errors, setErrors] = React.useState({});
  
  const handleChangeValue = fieldName => event => {
  
  const newForm = {...form};
    newForm[fieldName] =  event.target.value;
    setForm(newForm);
  };
  
  const handleSubmit = event => {
    event.preventDefault();
    const errs = validateForm(validationSetting, form);
    if (errs) {
      setErrors(errs);
      if(form.first_name===""||form.last_name===""||form.address1===""||form.address2===""||form.country===""||form.state===""||form.city===""){
        alert("fill your information form fully.");
      }
    }
    else {
       setRequestParams(form);
       history.push("/checkout/confirm");
    }
  };
    
  const addressForm = (
    <div>
      <form noValidate onSubmit={handleSubmit}>
        <div className="firstName">
          <TextField
            className={clsx(classes.textField, classes.dense)}
            id={"first_name"}
            name={"first_name"}
            value={form.first_name}
            error={errors.first_name ? true : false}
            required={validationSetting.isEmpty.indexOf("first_name") >=0 ? true : false}
            onChange={handleChangeValue("first_name")}
            label="First name"
            fullWidth
            autoComplete="fname"
            variant="outlined"
          />
        </div>
         
        <div className="lastName">
          <TextField
            className={clsx(classes.textField, classes.dense)}
            id={"last_name"}
            name={"last_name"}
            value={form.last_name}
            error={errors.last_name ? true : false}
            required={validationSetting.isEmpty.indexOf("last_name") >=0 ? true : false}
            onChange={handleChangeValue("last_name")}
            label="Last name"
            fullWidth
            autoComplete="lname"
            variant="outlined"
          />
        </div>
        
        <div className="address1">
          <TextField
            className={clsx(classes.textField, classes.dense)}
            id={"address1"}
            name={"address1"}
            value={form.address1}
            error={errors.address1 ? true : false}
            required={validationSetting.isEmpty.indexOf("address1") >=0 ? true : false}
            onChange={handleChangeValue("address1")}
            label="address1"
            fullWidth
            variant="outlined"
          />
        </div>
        
        <div className="address2">
          <TextField
           className={clsx(classes.textField, classes.dense)}
            id={"address2"}
            name={"address2"}
            value={form.address2}
            error={errors.address2 ? true : false}
            required={validationSetting.isEmpty.indexOf("address2") >=0 ? true : false}
            onChange={handleChangeValue("address2")}
            label="address2"
            fullWidth
            variant="outlined"
          />
        </div>
        
        <div className="country">
          <TextField
            className={clsx(classes.textField, classes.dense)}
            id={"country"}
            name={"country"}
            value={form.country}
            error={errors.country ? true : false}
            required={validationSetting.isEmpty.indexOf("country") >=0 ? true : false}
            onChange={handleChangeValue("country")}
            label="country"
            fullWidth
            variant="outlined"
          />
        </div>
        
        <div className="statetxt">
          <TextField
            className={clsx(classes.textField, classes.dense)}
            id={"state"}
            name={"state"}
            value={form.state}
            error={errors.state ? true : false}
            required={validationSetting.isEmpty.indexOf("state") >=0 ? true : false}
            onChange={handleChangeValue("state")}
            label="state"
            fullWidth
            variant="outlined"
          />
        </div>
        
        <div className="city">
          <TextField
            className={clsx(classes.textField, classes.dense)}
            id={"city"}
            name={"city"}
            value={form.city}
            error={errors.city ? true : false}
            required={validationSetting.isEmpty.indexOf("city") >=0 ? true : false}
            onChange={handleChangeValue("city")}
            label="city"
            fullWidth
            variant="outlined"
          />
        </div>
        
        <div className="createAccount">
          <Button
            type="submit"
            fullWidth
            variant="contained"
            color="secondary"
            onClick={handleSubmit}
            disabled={!cart.length}
           
          
          >
           <FormattedMessage id="Button.Confirm" defualtMessage="Confirm" />
          </Button>
        </div>
      </form>
    </div>
  );
  
  const handleHome = event => {
    event.preventDefault();
    history.push("/");
  };

  return (
    <React.Fragment>
       <TitleBar showMenu={false} showNav={false} showIcon={true} />
        <div className={classes.toolbar} />
        
        <div className="title_Ship">
        
          <div className="ho">1.<FormattedMessage id="Label.ShippingInformation" defualtMessage="Shipping Information" /></div>
          <div className="txt3"><FormattedMessage id="Label.Information" defualtMessage="Information" /></div>
          <div className="text2"><FormattedMessage id="Label.InformationDetail" defualtMessage="Please fill your information fully" /></div>
        
          <Grid>
            <Grid item xs={12} sm={12} md={8} lg={8}>
              <Box display="flex" >
                <Box ml="auto" my="auto" mr={1}>
                  <Button onClick={handleHome} variant="contained" color="primary">
                    <HomeIcon className={classes.homeSpace} />
                    <FormattedMessage id="Button.Shopping" defualtMessage="ShoppingContinuation" />
                  </Button>
                </Box>
              </Box>
              {addressForm}
              <Divider/> 
            </Grid>
          </Grid>
        </div>
        
        <div className="title_Ship">
          <div className="ho">2.<FormattedMessage id="Label.OrderReview" defualtMessage="Order Review" /></div>
           
           
           
          <Grid>
            <Grid item xs={12} sm={12} md={8} lg={8}>
              <CartTable showQty={true}/>
              <Box display="flex" flexDirection="column" marginTop="20px" bgcolor="#f2f2f2">
                <Paper className={classes.gridPaper}>
                  <Box  mt={1} mb={1}  textAlign="right" className="text1" mr={0} display="flex">
                    <Grid className="text1" item xs={9} sm={8}><FormattedMessage id="Label.TotalQuantity" defualtMessage="Total Quantity" />:</Grid> <Grid item xs={3} sm={4} className={classes.gridTotalPrice}> {totalQuantity}(Items)</Grid>
                  </Box>
                  <Box  mt={1} mb={1} textAlign="right" className="text1" mr={0} display="flex">
                    <Grid className="text1" item xs={9} sm={8}><FormattedMessage id="Label.TotalPrice" defualtMessage="Total Price" />:</Grid> <Grid item xs={3} sm={4} className={classes.gridTotalPrice}> {totalPrice}(MMK)</Grid>
                  </Box>
                </Paper>
              </Box>  
              <Divider/> 
           </Grid>
         </Grid>
        </div>
    </React.Fragment>
    
  );
};
CheckoutForm.propTypes = {
   totalPrice: PropTypes.number.isRequired,
   totalQuantity: PropTypes.number.isRequired,
};
export default withRouter(CheckoutForm);