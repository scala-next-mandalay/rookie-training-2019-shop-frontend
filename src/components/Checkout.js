import React from 'react';
import CheckoutForm from '../containers/CheckoutForm';
import CheckoutConfirm from '../containers/CheckoutConfirm';
import CheckoutSuccess from '../containers/CheckoutSuccess';

const Checkout = ({ showForm, cart, requestParams, postResultObj, clearCheckout }) => {
  let comp = <CheckoutForm />;
  if (showForm || cart.length === 0) {
    //Don't show old form data. So clear old form data. 
    clearCheckout();
  }
  else if (postResultObj != null) {
    comp = <CheckoutSuccess />;
  }
  else if (requestParams != null) {
    comp = <CheckoutConfirm />;
  }
  return (comp);
};

export default Checkout;