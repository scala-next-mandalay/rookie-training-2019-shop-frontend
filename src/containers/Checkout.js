import { connect } from 'react-redux';
import Checkout from '../components/Checkout';
import { clearCheckout } from '../modules/orders'

export default connect(
  (state) => ({
    cart: state.cart.rows,
    requestParams: state.orders.requestParams,
    postResultObj: state.orders.postResultObj,
  }),
  (dispatch) => ({
    clearCheckout: () =>  dispatch(clearCheckout())
  })
)(Checkout);

