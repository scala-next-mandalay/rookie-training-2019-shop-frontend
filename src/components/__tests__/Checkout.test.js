import React from 'react'
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import Checkout from '../Checkout';
import Parent from '../Parent';
import CheckoutForm from '../CheckoutForm';
/*global expect*/
/*global jest*/
let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})



afterEach(() => {
  document.body.outerHTML = '<body></body>'
  container = null
})

const valueSetter = Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, "value").set;
const requestParams = {
  first_name: "khin",
  last_name: "lawoon",
  address1: "23 street",
  address2: "CMTZ",
  country: "myanmar",
  state: "mandalay",
  city: "mandalay",
  total_price: 50000
}

const postResultObj = {
  id: 5,
  total_price: 50000
}

let showForm=true;
let comp=<CheckoutForm/>
const clearCheckout=jest.fn(()=>{
    
})
// const setRequestParams = jest.fn((form) => {
//   let setRequest ={...form};
// });

describe("Checkout component", () => {
  it('matches the snapshot', () => {
    const CheckoutSnapshot = renderer.create(<Parent>
        <Checkout showForm={showForm} requestParams={requestParams} postResultObj={postResultObj} clearCheckout={clearCheckout} />
    </Parent>).toJSON();

    expect(CheckoutSnapshot).toMatchSnapshot();

  })

})
describe("Testing Checkout Component", () => {
 it('testing ', () => {
    act(() => {
      ReactDOM.render((<Parent>
        <Checkout showForm={showForm} requestParams={requestParams} postResultObj={postResultObj} clearCheckout={clearCheckout} />
        </Parent>), container);
        
   
    });
    expect(clearCheckout).toHaveBeenCalledWith();
     
 });
});