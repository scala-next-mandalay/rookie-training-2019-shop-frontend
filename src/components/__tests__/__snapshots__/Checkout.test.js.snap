// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout component matches the snapshot 1`] = `
Array [
  <div>
    <header
      className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-13 MuiAppBar-colorPrimary mui-fixed"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="Open drawer"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-14 MuiIconButton-edgeStart"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          />
        </button>
        <div
          className="PrivateHiddenCss-smDown-79"
        >
          <div
            className="MuiBox-root MuiBox-root-90 title"
          >
            <span>
              Rookie Training 2019
            </span>
          </div>
        </div>
        <div
          className="MuiBox-root MuiBox-root-91"
        >
          <div
            className="MuiGrid-root MuiGrid-container"
          >
            <div
              className="MuiBox-root MuiBox-root-195 makeStyles-navMenuList-18"
            />
            <button
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary Mui-disabled Mui-disabled"
              disabled={true}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <span
                  className="MuiBadge-root"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                    />
                  </svg>
                  <span
                    className="MuiBadge-badge MuiBadge-colorSecondary MuiBadge-invisible"
                  >
                    0
                  </span>
                </span>
              </span>
            </button>
            <button
              aria-controls="long-menu"
              aria-haspopup="true"
              aria-label="More"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </header>
  </div>,
  <div
    className="makeStyles-toolbar-11"
  />,
  <div
    className="title_Ship"
  >
    <div
      className="ho"
    >
      1.
      <span>
        Shipping Information
      </span>
    </div>
    <div
      className="txt3"
    >
      <span>
        Information
      </span>
    </div>
    <div
      className="text2"
    >
      <span>
        Please fill your information fully
      </span>
    </div>
    <div
      className="MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-8 MuiGrid-grid-lg-8"
      >
        <div
          className="MuiBox-root MuiBox-root-227"
        >
          <div
            className="MuiBox-root MuiBox-root-228"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root makeStyles-homeSpace-12"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                  />
                </svg>
                <span>
                  Shopping
                </span>
              </span>
            </button>
          </div>
        </div>
        <div>
          <form
            noValidate={true}
            onSubmit={[Function]}
          >
            <div
              className="firstName"
            >
              <div
                className="MuiFormControl-root MuiTextField-root makeStyles-textField-8 makeStyles-dense-9 MuiFormControl-fullWidth"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                  data-shrink={false}
                  htmlFor="first_name"
                >
                  First name
                  <span
                    className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                    â€‰
                    *
                  </span>
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoComplete="fname"
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    id="first_name"
                    name="first_name"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-303 MuiOutlinedInput-notchedOutline"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="PrivateNotchedOutline-legend-304"
                      style={
                        Object {
                          "width": 0.01,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              className="lastName"
            >
              <div
                className="MuiFormControl-root MuiTextField-root makeStyles-textField-8 makeStyles-dense-9 MuiFormControl-fullWidth"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink={false}
                  htmlFor="last_name"
                >
                  Last name
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoComplete="lname"
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    id="last_name"
                    name="last_name"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-303 MuiOutlinedInput-notchedOutline"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="PrivateNotchedOutline-legend-304"
                      style={
                        Object {
                          "width": 0.01,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              className="address1"
            >
              <div
                className="MuiFormControl-root MuiTextField-root makeStyles-textField-8 makeStyles-dense-9 MuiFormControl-fullWidth"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                  data-shrink={false}
                  htmlFor="address1"
                >
                  address1
                  <span
                    className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                    â€‰
                    *
                  </span>
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    id="address1"
                    name="address1"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-303 MuiOutlinedInput-notchedOutline"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="PrivateNotchedOutline-legend-304"
                      style={
                        Object {
                          "width": 0.01,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              className="address2"
            >
              <div
                className="MuiFormControl-root MuiTextField-root makeStyles-textField-8 makeStyles-dense-9 MuiFormControl-fullWidth"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink={false}
                  htmlFor="address2"
                >
                  address2
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    id="address2"
                    name="address2"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-303 MuiOutlinedInput-notchedOutline"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="PrivateNotchedOutline-legend-304"
                      style={
                        Object {
                          "width": 0.01,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              className="country"
            >
              <div
                className="MuiFormControl-root MuiTextField-root makeStyles-textField-8 makeStyles-dense-9 MuiFormControl-fullWidth"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                  data-shrink={false}
                  htmlFor="country"
                >
                  country
                  <span
                    className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                    â€‰
                    *
                  </span>
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    id="country"
                    name="country"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-303 MuiOutlinedInput-notchedOutline"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="PrivateNotchedOutline-legend-304"
                      style={
                        Object {
                          "width": 0.01,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              className="statetxt"
            >
              <div
                className="MuiFormControl-root MuiTextField-root makeStyles-textField-8 makeStyles-dense-9 MuiFormControl-fullWidth"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink={false}
                  htmlFor="state"
                >
                  state
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    id="state"
                    name="state"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-303 MuiOutlinedInput-notchedOutline"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="PrivateNotchedOutline-legend-304"
                      style={
                        Object {
                          "width": 0.01,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              className="city"
            >
              <div
                className="MuiFormControl-root MuiTextField-root makeStyles-textField-8 makeStyles-dense-9 MuiFormControl-fullWidth"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
                  data-shrink={false}
                  htmlFor="city"
                >
                  city
                  <span
                    className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                    â€‰
                    *
                  </span>
                </label>
                <div
                  className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    className="MuiInputBase-input MuiOutlinedInput-input"
                    disabled={false}
                    id="city"
                    name="city"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={true}
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden={true}
                    className="PrivateNotchedOutline-root-303 MuiOutlinedInput-notchedOutline"
                    style={
                      Object {
                        "paddingLeft": 8,
                      }
                    }
                  >
                    <legend
                      className="PrivateNotchedOutline-legend-304"
                      style={
                        Object {
                          "width": 0.01,
                        }
                      }
                    >
                      <span
                        dangerouslySetInnerHTML={
                          Object {
                            "__html": "&#8203;",
                          }
                        }
                      />
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <div
              className="createAccount"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary Mui-disabled MuiButton-fullWidth Mui-disabled"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                type="submit"
              >
                <span
                  className="MuiButton-label"
                >
                  <span>
                    Confirm
                  </span>
                </span>
              </button>
            </div>
          </form>
        </div>
        <hr
          className="MuiDivider-root"
        />
      </div>
    </div>
  </div>,
  <div
    className="title_Ship"
  >
    <div
      className="ho"
    >
      2.
      <span>
        Order Review
      </span>
    </div>
    <div
      className="MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-8 MuiGrid-grid-lg-8"
      >
        <div
          className="PrivateHiddenCss-smUp-78"
        />
        <div
          className="PrivateHiddenCss-xsDown-76"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              className="makeStyles-girdDv-315"
            >
              <div
                className="MuiGrid-root makeStyles-gridRow-310 MuiGrid-container"
                variant="contained"
              >
                <div
                  className="MuiGrid-root makeStyles-gridCell-311 MuiGrid-item MuiGrid-grid-xs-3"
                >
                  <div
                    className="MuiBox-root MuiBox-root-316"
                  >
                    <span>
                      Image
                    </span>
                  </div>
                </div>
                <div
                  className="MuiGrid-root makeStyles-gridCell-311 MuiGrid-item MuiGrid-grid-xs-3"
                >
                  <div
                    className="MuiBox-root MuiBox-root-317"
                  >
                    <span>
                      Description
                    </span>
                  </div>
                </div>
                <div
                  className="MuiGrid-root makeStyles-gridCell-311 MuiGrid-item MuiGrid-grid-xs-2"
                >
                  <div
                    className="MuiBox-root MuiBox-root-318"
                  >
                    <span>
                      Price
                    </span>
                  </div>
                </div>
                <div
                  className="MuiGrid-root makeStyles-gridCell-311 MuiGrid-item MuiGrid-grid-xs-2"
                >
                  <div
                    className="MuiBox-root MuiBox-root-319"
                  >
                    <span>
                      Quantity
                    </span>
                  </div>
                </div>
                <div
                  className="MuiGrid-root makeStyles-gridCell-311 MuiGrid-item MuiGrid-grid-xs-2"
                >
                  <div
                    className="MuiBox-root MuiBox-root-320"
                  >
                    <span>
                      Total Price
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
            >
              <hr
                className="MuiDivider-root"
              />
            </div>
          </div>
        </div>
        <div
          className="MuiBox-root MuiBox-root-321"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation1 makeStyles-gridPaper-6 MuiPaper-rounded"
          >
            <div
              className="MuiBox-root MuiBox-root-322 text1"
            >
              <div
                className="MuiGrid-root text1 MuiGrid-item MuiGrid-grid-xs-9 MuiGrid-grid-sm-8"
              >
                <span>
                  Total Quantity
                </span>
                :
              </div>
               
              <div
                className="MuiGrid-root makeStyles-gridTotalPrice-7 MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-sm-4"
              >
                 
                0
                (Items)
              </div>
            </div>
            <div
              className="MuiBox-root MuiBox-root-323 text1"
            >
              <div
                className="MuiGrid-root text1 MuiGrid-item MuiGrid-grid-xs-9 MuiGrid-grid-sm-8"
              >
                <span>
                  Total Price
                </span>
                :
              </div>
               
              <div
                className="MuiGrid-root makeStyles-gridTotalPrice-7 MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-sm-4"
              >
                 
                0
                (MMK)
              </div>
            </div>
          </div>
        </div>
        <hr
          className="MuiDivider-root"
        />
      </div>
    </div>
  </div>,
]
`;
